RELEASE.md

```diff
@@ -0,0 +1,20 @@
+# Release instructions
+
+Steps to release a new version of the OpenTelemetry Operator:
+
+1. Change the `versions.txt`, so that it lists the target version of the OpenTelemetry Collector (operand), and the desired version for the operator. The `major.minor` should typically match, with the patch portion being possibly different.
```

Comment:
Why "desired version for the operator"? Isn't it *the* version of the operator.

Reply:
This is a change compared to the Jaeger Operator: we don't have an intermediary step now where we bump the versions. So, the version bumping is done before the release is tagged.